Make "./mach install" use the correct directory/symlink names.

--- a/config/baseconfig.mk
+++ b/config/baseconfig.mk
@@ -2,7 +2,7 @@
 # directly in python/mozbuild/mozbuild/base.py for gmake validation.
 # We thus use INCLUDED_AUTOCONF_MK to enable/disable some parts depending
 # whether a normal build is happening or whether the check is running.
-installdir = $(libdir)/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_PKG_APPNAME)
 includedir := $(includedir)/$(MOZ_APP_NAME)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
--- a/toolkit/mozapps/installer/packager.mk
+++ b/toolkit/mozapps/installer/packager.mk
@@ -173,7 +173,7 @@ endif
 	  (cd $(DESTDIR)$(installdir) && tar -xf -)
 	$(NSINSTALL) -D $(DESTDIR)$(bindir)
 	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
-	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
+	ln -s $(installdir)/$(MOZ_PKG_APPNAME) $(DESTDIR)$(bindir)
 
 upload:
 	$(PYTHON3) -u $(MOZILLA_DIR)/build/upload.py --base-path $(DIST) $(UPLOAD_FILES)
