# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  557d942b99754b10e0d154749c84f849e0b81747
Bug 532712 - disable some code that replaces NBSP with ASCII space.

diff --git a/editor/libeditor/WSRunObject.cpp b/editor/libeditor/WSRunObject.cpp
--- a/editor/libeditor/WSRunObject.cpp
+++ b/editor/libeditor/WSRunObject.cpp
@@ -942,23 +942,25 @@ nsresult WhiteSpaceVisibilityKeeper::Rep
       if (!atNBSPReplacedWithASCIIWhiteSpace.IsEndOfContainer() &&
           atNBSPReplacedWithASCIIWhiteSpace.IsCharNBSP()) {
         AutoTrackDOMPoint trackPointToInsert(aHTMLEditor.RangeUpdaterRef(),
                                              &pointToInsert);
         AutoTrackDOMRange trackInvisibleTrailingWhiteSpaceRange(
             aHTMLEditor.RangeUpdaterRef(),
             &invisibleTrailingWhiteSpaceRangeAtEnd);
         AutoTransactionsConserveSelection dontChangeMySelection(aHTMLEditor);
+        /*
         nsresult rv = aHTMLEditor.ReplaceTextWithTransaction(
             MOZ_KnownLive(*atNBSPReplacedWithASCIIWhiteSpace.ContainerAsText()),
             atNBSPReplacedWithASCIIWhiteSpace.Offset(), 1, u" "_ns);
         if (NS_FAILED(rv)) {
           NS_WARNING("HTMLEditor::ReplaceTextWithTransaction() failed failed");
           return rv;
         }
+        */
         // Don't refer the following variables anymore unless tracking the
         // change.
         atNBSPReplaceableWithSP.Clear();
         invisibleLeadingWhiteSpaceRangeAtStart.Clear();
       }
     }
   }
 
@@ -2881,27 +2883,29 @@ nsresult WhiteSpaceVisibilityKeeper::Nor
               atPreviousCharOfPreviousCharOfEndOfVisibleWhiteSpaces
                   .IsCharASCIISpace();
           followedByVisibleContentOrBRElement = true;
         }
       }
 
       // Next, replace the NBSP with an ASCII white-space if it's surrounded
       // by visible contents (or immediately before a <br> element).
+      /*
       if (maybeNBSPFollowingVisibleContent &&
           followedByVisibleContentOrBRElement) {
         AutoTransactionsConserveSelection dontChangeMySelection(aHTMLEditor);
         nsresult rv = aHTMLEditor.ReplaceTextWithTransaction(
             MOZ_KnownLive(
                 *atPreviousCharOfEndOfVisibleWhiteSpaces.ContainerAsText()),
             atPreviousCharOfEndOfVisibleWhiteSpaces.Offset(), 1, u" "_ns);
         NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
                              "HTMLEditor::ReplaceTextWithTransaction() failed");
         return rv;
       }
+      */
     }
     // If the text node is not preformatted, and the NBSP is followed by a <br>
     // element and following (maybe multiple) ASCII spaces, remove the NBSP,
     // but inserts a NBSP before the spaces.  This makes a line break
     // opportunity to wrap the line.
     // XXX This is different behavior from Blink.  Blink generates pairs of
     //     an NBSP and an ASCII white-space, but put NBSP at the end of the
     //     sequence.  We should follow the behavior for web-compat.
