# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  5d0f2f80b97316377e7d6c85546f0699b444e826
Bug 1776580 - Fix alert for multiple accounts: Only set biff after really showing the alert.

diff --git a/mailnews/base/src/MailNotificationManager.jsm b/mailnews/base/src/MailNotificationManager.jsm
--- a/mailnews/base/src/MailNotificationManager.jsm
+++ b/mailnews/base/src/MailNotificationManager.jsm
@@ -355,18 +355,16 @@ class MailNotificationManager {
         this._logger.error(e);
         this._systemAlertAvailable = false;
       }
     }
 
     // The use_system_alert pref is false or showAlert somehow failed, use the
     // customized alert window.
     this._showCustomizedAlert(folder);
-
-    this._folderBiffTime.set(folder, Date.now());
   }
 
   /**
    * Show a customized alert window (newmailalert.xhtml), if there is already
    * one showing, do not show another one, because the newer one will block the
    * older one. Instead, save the folder and newMsgKeys to this._pendingFolders.
    * @param {nsIMsgFolder} [folder] - The folder containing new messages.
    */
@@ -406,16 +404,17 @@ class MailNotificationManager {
     Services.ww.openWindow(
       null,
       "chrome://messenger/content/newmailalert.xhtml",
       "_blank",
       "chrome,dialog=yes,titlebar=no,popup=yes",
       args
     );
     this._customizedAlertShown = true;
+    this._folderBiffTime.set(folder, Date.now());
   }
 
   /**
    * Get all NEW messages from a folder that we received after last biff time.
    * @param {nsIMsgFolder} folder - The message folder to check.
    * @returns {number[]} An array of message keys.
    */
   _getNewMsgKeysNotNotified(folder) {
